kpi/filters.py

```diff
@@ -219,3 +220,45 @@ def filter_queryset(self, request, queryset, view):
                 ).values_list('pk', flat=True)
             )
         return queryset.filter(pk__in=object_permission_ids)
+
+
+class AttachmentFilter(filters.BaseFilterBackend):
+    def filter_queryset(self, request, queryset, view):
+        type = request.query_params.get('type')
```

Comment:
we should avoid this variable name since it overlaps with python's builtin `type()`. Python convention seems to be to add a trailing underscore; `type_`.